director/accounts/views.py

```diff
@@ -73,16 +76,34 @@ def has_any_permissions(
 
         return False
 
+    def is_permitted(
+        self,
+        user: User,
+        permission: AccountPermissionType,
+        account_name: typing.Optional[str] = None,
+        pk: typing.Optional[int] = None,
+    ) -> bool:
+        """Fetch an `Account` using its `pk` or `name`, then determine if `user` has `permission` for the `Account`."""
+        self.perform_account_fetch(user, account_name, pk)
+        return self.has_permission(permission)
+
     def request_permissions_guard(
```

Comment:
This method existed before but I've made it better. I think this should be the preferred use for guarding against incorrect permissions. It also does the account fetch so you can refer to `self.account` later in views once this has been called.

By default it will check for `self.required_account_permission` but you can override this by passing in a permission. This way we can have some of the view methods requiring `VIEW` and editing methods requiring `ADMINISTER` for example.